### PR TITLE
Fix spatially varying boundary conditions on GPU

### PR DESCRIPTION
**Pull request by simonbyrne**
## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by szy21** [comment]
Try use `kappa_4: 1.0e15`

**Comment by sriharshakandala** [comment]
Can we remove the `soft_fail: true` flag in the CI pipeline!

File: config/gpu_configs/gpu_hs_rhoe_equilmoist_nz63_0M_55km_rs35km.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
 dt_save_to_disk: "10days"
-dt: "150secs"
-t_end: "300days"
-h_elem: 16
+dt: "100secs"
+t_end: "1days"
```
Comment (charleskawczynski) [inline config/gpu_configs/gpu_hs_rhoe_equilmoist_nz63_0M_55km_rs35km.yml:3]: I suggest we go at least 10 days, so that the callbacks are exercised.

Comment (simonbyrne) [reply config/gpu_configs/gpu_hs_rhoe_equilmoist_nz63_0M_55km_rs35km.yml:3]: what callbacks need to be exercised?

**Review by sriharshakandala — APPROVED** [review]

File: config/gpu_configs/gpu_hs_rhoe_equilmoist_nz63_0M_55km_rs35km.yml (hunk 1 above)
Comment (charleskawczynski) [reply config/gpu_configs/gpu_hs_rhoe_equilmoist_nz63_0M_55km_rs35km.yml:3]: The default are probably fine, which will probably include a bunch of model-specific pieces (e.g., diagnostics for moisture etc.).

**Comment by simonbyrne** [comment]
> Can we remove the `soft_fail: true` flag in the CI pipeline!

Done

File: config/gpu_configs/gpu_hs_rhoe_equilmoist_nz63_0M_55km_rs35km.yml (hunk 1 above)
Comment (simonbyrne) [reply config/gpu_configs/gpu_hs_rhoe_equilmoist_nz63_0M_55km_rs35km.yml:3]: can we just reduce the `dt_save_to_disk`?

Comment (simonbyrne) [reply config/gpu_configs/gpu_hs_rhoe_equilmoist_nz63_0M_55km_rs35km.yml:3]: okay, but that doesn't need to be in the perf target. We should make sure it is in the CI though.

Comment (charleskawczynski) [reply config/gpu_configs/gpu_hs_rhoe_equilmoist_nz63_0M_55km_rs35km.yml:3]: > can we just reduce the dt_save_to_disk?

Yeah, that works too

**Pull request merged**